build_tools/aws-sdk-code-generator/spec/protocols_spec.rb

```diff
@@ -106,7 +106,11 @@ def format_data(ref, src)
       params[key] = format_data(ref.shape.value, value)
     end
   when Seahorse::Model::Shapes::TimestampShape
-    Time.at(src)
+    if src.is_a?(String)
```

Comment:
I'd lean towards using `case` here - and to preserve the previous behavior, doing `Time.at` on the else.  Something like:
```
case src
when String
  Time.parse(src)
else
  Time.at(src)
end  
```